data-structures-and-algorithms/lesson02.rst

```diff
@@ -4,6 +4,22 @@ Data Structures and Algorithm Analysis - Lesson 2 Learning Objectives
 Goals
 -----
 
+- Use binary search to efficiently find elements in ordered collections as well as binary search trees
+- Understand Binary Search Tree operations and their big-O efficiency
 
 Objectives
 ----------
+
+- Understand and write code to conduct a binary search on an ordered collection (e.g. an array or list)
+- Explain why a data set must be sorted before conducting a binary search
+- Understand the conceptual structure of a binary tree
+- Understand the conceptual structure of a binary search tree
+- Explain how to carry out binary search on a binary search tree
```

Comment:
This is in reference to the algorithm to search for an element in a binary tree. More specifically:

1. Compare search item to root element. 
2. If the item equals the root, we found it.
3. Go left if it's less, right if it's greater.
4. Recursively search the subtree until the element is found or a leaf is reached.

Reply:
thank you that was a really clear explanation.

i meant should we add 

> Understand how a binary tree can be made into binary search tree

or 

> Understand the differences between a BT and BST

Reply:
Got it. Converting a binary tree to a BST isn't something that is commonly done. Since a BST is considered to be sorted, it won't have a sort method. So any such conversion would be to extract all elements from the binary tree into an array, and then build a BST from the elements of that array. In all but exceptional cases, however, you'll know you want stuff sorted and so will start with a BST.

Reply:
interesting. so what is the use of a BT itself then? just as a data structure for tree like data (DOM for example) but without the search optimization?